galsim/bandpass.py

```diff
@@ -463,12 +430,49 @@ def thin(self, rel_err=1.e-4, preserve_range=False):
             f = self(x)
             newx, newf = utilities.thin_tabulated_values(x, f, rel_err=rel_err,
                                                          preserve_range=preserve_range)
-            # preserve type
-            ret = self.copy()
-            ret.func = galsim.LookupTable(newx, newf, interpolant='linear')
-            ret.blue_limit = np.min(newx)
-            ret.red_limit = np.max(newx)
-            ret.wave_list = np.array(newx)
-            if hasattr(ret, '_effective_wavelength'):
-                del ret._effective_wavelength
-            return ret
+            tp = galsim.LookupTable(newx, newf, interpolant='linear')
+            blue_limit = np.min(newx)
+            red_limit = np.max(newx)
+            wave_list = np.array(newx)
+            return Bandpass(tp, blue_limit, red_limit, _wave_list=wave_list)
+        else:
+            return self
+
+    def __eq__(self, other):
+        return (isinstance(other, Bandpass) and
+                self._orig_tp == other._orig_tp and
+                self.red_limit == other.red_limit and
+                self.blue_limit == other.blue_limit and
+                self.wave_factor == other.wave_factor and
+                self.zeropoint == other.zeropoint and
+                (self.wave_list == other.wave_list).all())
+
+    def __getstate__(self):
+        d = self.__dict__.copy()
+        del d['_tp']
```

Comment:
Good point.  `func` needs to be deleted, since that's always a lambda expression.  But I changed it so `_tp` (and `_spec' in SED) is only deleted if it is not a LookupTable.  As you point out, these are picklable directly, and it's probably more efficient to do so rather than read them in from the file again.